Marlin/src/HAL/STM32/tft/gt911.h

```diff
@@ -0,0 +1,120 @@
+/**
+ * Marlin 3D Printer Firmware
+ * Copyright (c) 2020 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (c) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+#pragma once
+
+#include "../../../inc/MarlinConfig.h"
+
+#define GT911_SLAVE_ADDRESS   0xBA
+
+#if !PIN_EXISTS(GT911_RST)
+  #error "GT911_RST_PIN is not defined."
+#elif !PIN_EXISTS(GT911_INT)
+  #error "GT911_INT_PIN is not defined."
+#elif !PIN_EXISTS(GT911_SW_I2C_SCL)
+  #error "GT911_SW_I2C_SCL_PIN is not defined."
+#elif !PIN_EXISTS(GT911_SW_I2C_SDA)
+  #error "GT911_SW_I2C_SDA_PIN is not defined."
+#endif
+
+class SW_IIC {
```

Comment:
Marlin doesn't already have a IIC class? if not, I think it should to some of the core folders (like features or mudules)

Reply:
I googled the code and found BL24CXX has a little IIC interface based on `libmaple/gpio.h`. The new little software IIC in the GT911 code is self-contained, so might be a good choice.

I see that the GT911 class for IIC is using `OUT_WRITE(scl_pin, HIGH)` with a variable. That won't work under some HALs … only those that pass through to `pinMode` and `digitalWrite`.